dump_nodes.py

```
@@ -0,0 +1,107 @@
+                group = node_type.parmTemplateGroup()
+                parms = extract_parms(group.entries())
+            except Exception as e:
+                logger.debug(f"Could not extract parameters for node '{node_name}' in '{cat_name}': {e}")
+
+            try:
+                cat_data[node_name] = {
+                    "min_inputs": node_type.minNumInputs(),
+                    "max_inputs": node_type.maxNumInputs(),
+                    "parms": parms
+                }
+            except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)